bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/util/TracingUtilities.java

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * These utilities are to be used in conjunction with This is to be used in conjunction with the
+ * opencensus-contrib-zpages artifact. After including the artifact, the following code can be used
+ * for enabling grpcz pages:
+ * 
+ * <pre>
+ * try {
+ *   TracingUtlities.setupTracingConfig();
+ *   // This call is optional if you have a port you want to use
+ *   int port = TracingUtilities.getFreePort();
+ *   ZPageHandlers.startHttpServerAndRegisterAll(port);
+ * } catch (Throwable e) {
+ *   LOG.error("Could not initialize ZPageHandlers", e);
+ * }
+ * </pre>
+ * 
+ * This is a method of enabling exports to stackdriver along with the
+ * opencensus-exporter-trace-stackdriver artifact:
+ * 
+ * <pre>
+ * try {
+ *   StackdriverExporter.createAndRegister(projectId);
+ * } catch (Throwable e) {
+ *   LOG.error("Could not register stackdriver", e);
+ * }
+ * </pre>
+ */
+public final class TracingUtilities {
+
+  /**
+   * Finds a free port that can be used as the grpcz page port.
+   *
+   * @return a free port nubmer.
+   * @throws IOException
+   */
+  public static int getFreePort() throws IOException {
+    try (ServerSocket s = new ServerSocket(0)) {
+      return s.getLocalPort();
+    } catch (IOException e) {
+      throw new IOException("Failed to find a free port", e);
+    }
+  }
+
+  /**
+   * This is a one time setup for grpcz pages. This adds all of the methods to the Tracing
+   * environment required to show a consistent set of methods relating to Cloud Bigtable on the
+   * grpcz page.  If HBase artifacts are present, this will add tracing metadata for HBase methods.
+   */
+  public static void setupTracingConfig() {
+    List<String> descriptors = new ArrayList<>();
+    addDescriptor(descriptors, BigtableTableAdminGrpc.getServiceDescriptor());
+    addDescriptor(descriptors, BigtableGrpc.getServiceDescriptor());
+
+    boolean usingBigtableHBase = false;
+    try {
+      Class.forName("com.google.cloud.bigtable.hbase.BigtableTable");
+      usingBigtableHBase = true;
+    } catch (ClassNotFoundException e) {
+    }
+    if (usingBigtableHBase) {
+      descriptors.addAll(Arrays.asList(
+        "BigtableTable.getTableDescriptor",
+        "BigtableTable.exists",
+        "BigtableTable.existsAll",
+        "BigtableTable.batch",
+        "BigtableTable.batchCallback",
+        "BigtableTable.get",
+        "BigtableTable.put",
+        "BigtableTable.checkAndPut",
+        "BigtableTable.delete",
+        "BigtableTable.checkAndDelete",
+        "BigtableTable.checkAndMutate",
+        "BigtableTable.mutateRow",
+        "BigtableTable.append",
+        "BigtableTable.increment",
+        "BigtableTable.incrementColumnValue"
+      ));
+    }
+
+    Tracing.getExportComponent().getSampledSpanStore().registerSpanNamesForCollection(descriptors);
+    Tracing.getTraceConfig().updateActiveTraceParams(
+      TraceParams.DEFAULT.toBuilder().setSampler(Samplers.probabilitySampler(0.1)).build());
```

Comment:
I don't think a component of a system (bigtable client) should configure global TraceParams

Reply:
0.1 is very high. You can leave it as 1/10000 and tell people to use /traceconfigz to change that.

Reply:
I didn't intend to check this in.  I'll remove it.